File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for(int i = 0; i< numRows ;i++){
            List<Integer> temp = new ArrayList<>();
            for(int j = 0; j < i+1; j++){
                if(j == 0 || j == i){
                    temp.add(1);
                }else{
                    List<Integer> prev =res.get(i-1);
                    int sum = prev.get(j - 1) + prev.get(j);
                    temp.add(sum);
                }
            }
            res.add(temp);
        }
        return res;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        l = []
        for i in range(numRows):
            temp = []
            for j in range(i+1):
                if j == 0 or j == i:
                    temp.append(1)
                else:
                    temp.append(l[i-1][j-1] + l[i-1][j])
            l.append(temp)
        return l